### ISSUE TITLE
Mismatch errors prevent html_report from caching via CI/CD

### ISSUE THREAD

**Issue by ViperART** [garris/BackstopJS#1220]
Greetings!

I'm trying to include backstop tests into CI (gitlab CI in particular). The idea is to host html report on gitlab pages, which is achieved (in my case) through caching backstop_data folder and then using this cache for hosting gitlab pages in next job.

Here's the job from .gitlab-ci.yml that creates report and caches results:

```
test:  
  stage: test
  image:
    name: backstopjs/backstopjs
    entrypoint: [""]
  script:
    - cd frontend/landings/landing
    - backstop reference
    - backstop test
  artifacts:
    when: always
    paths:
      - frontend/landings/landing/backstop_data
  cache:
    key: backstopData
    paths:
      - frontend/landings/landing/backstop_data
  when: manual
```

And it actually works properly - but only if tests are 100% successful (when all of my tests were successful I was able to see report on pages). But if one of the test cases fails, CLI process exits and cache is never written. Meanwhile, artifacts are working properly, and if I download them I can access HTML report just fine.

If I add fail proofing, such as `backstop reference || echo "test failed"`, bitmaps_reference and bitmaps_test folders make it to the cache, but still no access to html_report.

Is there a way to prevent backstop from killing a job on mismatch error? Or am I going in a wrong direction and there's a better way of hosting report on gitlab pages?

**Comment by garris** [comment]
What does the log show?

**Comment by ViperART** [comment]
```
report | Test completed...
        report | 17 Passed
        report | 3 Failed
        report | Writing browser report
        report | Resources copied
        report | Copied json report to: /builds/frontend/global/landing/frontend/landings/landing/backstop_data/bitmaps_test/20200828-151850/report.json
        report | Copied jsonp report to: /builds/frontend/global/landing/frontend/landings/landing/backstop_data/html_report/config.js
       COMMAND | Executing core for "openReport"
    openReport | Attempting to ping 
    openReport | Remote not found. Opening backstop_data/html_report/index.html
        report | *** Mismatch errors found ***
       COMMAND | Command "report" ended with an error after [1.242s]
       COMMAND | Error: Mismatch errors found.
                     at /usr/local/lib/node_modules/backstopjs/core/command/report.js:189:17
                     at process._tickCallback (internal/process/next_tick.js:68:7)
       COMMAND | Command "test" ended with an error after [19.383s]
       COMMAND | Error: Mismatch errors found.
                     at /usr/local/lib/node_modules/backstopjs/core/command/report.js:189:17
                     at process._tickCallback (internal/process/next_tick.js:68:7)
Uploading artifacts for failed job
 Uploading artifacts...
 frontend/landings/landing/backstop_data: found 145 matching files and directories 
 Uploading artifacts as "archive" to coordinator... ok  id=852089 responseStatus=201 Created token=EZdDnit_
 ERROR: Job failed: exit code 1
```

**Comment by ViperART** [comment]
any ideas?

**Comment by garris** [comment]
Sorry.  It's hard to understand what is going on here.  It is sounding like a race condition -- but I don't have Gitlab CI experience so I am not sure.   I think though that you should disable the auto openReport -- maybe that is what's erroring out?  Check the readme for how to do that.

**Comment by ViperART** [comment]
Thanks for the replies! In the end I managed to host html reports on gitlab pages.

For anyone who might be wondering how to do it - the problem indeed was with cache not being written because of job ending right after tests. As I told before, artifacts were written perfectly fine, but I thought I couldn't pass them between jobs. But it is possible - all I had to do is move 'test' and 'pages' jobs to different stages and add `dependencies: test` option to pages job, which allowed this job to use artifacts from 'test'.

Thanks again for the concern, I'm closing the issue.
